src/core/execute.c

```diff
@@ -2103,20 +2102,17 @@ static int setup_exec_directory(
                         if (r < 0)
                                 goto fail;
 
-                        effective = pp;
-
+                        /* Lock down the access mode */
+                        if (chmod(pp, context->directories[type].mode) < 0) {
+                                r = -errno;
+                                goto fail;
+                        }
                 } else {
                         r = mkdir_label(p, context->directories[type].mode);
-                        if (r < 0 && r != -EEXIST)
+                        if (r == -EEXIST)
+                                return 0;
```

Comment:
this is not right. there might be multiple directories configured, and we should definitely create any other directories even if the first one already exists, hence we should continue in this loop, and not return from the function altogether here

Reply:
Thanks, I don't know how I missed that.